app/vmui/packages/vmui/src/pages/ExploreAlerts/ExploreRules.tsx

```diff
@@ -0,0 +1,266 @@
+import { FC, useCallback, useEffect, useMemo, useState } from "preact/compat";
+import { useLocation, useSearchParams } from "react-router";
+import { useRulesSetQueryParams as useSetQueryParams } from "./hooks/useSetQueryParams";
+import Spinner from "../../components/Main/Spinner/Spinner";
+import Alert from "../../components/Main/Alert/Alert";
+import Accordion from "../../components/Main/Accordion/Accordion";
+import { useFetchGroups } from "./hooks/useFetchGroups";
+import "./style.scss";
+import RulesHeader from "../../components/ExploreAlerts/RulesHeader";
+import GroupHeader from "../../components/ExploreAlerts/GroupHeader";
+import Rule from "../../components/ExploreAlerts/Rule";
+import ExploreRule from "../../pages/ExploreAlerts/ExploreRule";
+import ExploreAlert from "../../pages/ExploreAlerts/ExploreAlert";
+import { Group, Rule as APIRule } from "../../types";
+import { getQueryStringValue } from "../../utils/query-string";
+import { getState, getChanges } from "./helpers";
+
+const defaultTypesStr = getQueryStringValue("types", "") as string;
+const defaultTypes = defaultTypesStr.split("&").filter((rt) => rt) as string[];
+const defaultStatesStr = getQueryStringValue("states", "") as string;
+const defaultStates = defaultStatesStr.split("&").filter((s) => s) as string[];
+const defaultSearchInput = getQueryStringValue("search", "") as string;
+
+interface ExploreRulesProps {
+  typeFilter: string;
+}
+
+const ExploreRules: FC<ExploreRulesProps> = ({ typeFilter }) => {
+  const groupId = getQueryStringValue("group_id", "") as string;
+  const ruleId = getQueryStringValue("rule_id", "") as string;
+  const alertId = getQueryStringValue("alert_id", "") as string;
+
+  const [expandGroups, setExpandGroups] = useState<Set<string>>(
+    new Set<string>(),
+  );
+  const [expandRules, setExpandRules] = useState<Set<string>>(
+    new Set<string>(),
+  );
+  const [searchInput, setSearchInput] = useState(defaultSearchInput);
+  const [types, setTypes] = useState(defaultTypes);
+  const [states, setStates] = useState(defaultStates);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const { hash } = useLocation();
+
+  if (!hash && groupId && (ruleId || alertId)) {
+    setModalOpen(true);
+  } else {
+    setModalOpen(false);
+  }
```

Comment:
It's better to move inside the `useEffect`
```suggestion
  useEffect(() => {
  if (!hash && groupId && (ruleId || alertId)) {
    setModalOpen(true);
  } else {
    setModalOpen(false);
  }
}, [hash, groupId, ruleId, alertId]);
```